rama-http/src/service/fs/serve_dir/open_file.rs

```diff
@@ -41,26 +65,95 @@ pub(super) struct FileOpened {
 pub(super) enum FileRequestExtent {
     Full(File, Metadata),
     Head(Metadata),
+    Embedded(Vec<u8>, u64), // Content, original file size
+    EmbeddedHead(u64),      // Content length for HEAD requests on embedded files
+}
+
+impl FileRequestExtent {
+    pub(super) fn reader(self) -> Option<impl AsyncRead + Send + Sync + Unpin> {
```

Comment:
```suggestion
    pub(super) fn into_reader(self) -> Option<impl AsyncRead + Send + Sync + Unpin> {
```

to make it clear we are turning `self` into something else. Because when I read `reader` alone I assume it is not consuming, while here it is.